### PR TITLE
fix: Improve message event handling when there are several runtime bundles

### PR DESCRIPTION
**Pull request by igdianov**
deployed using the same process definition but sharing one Rabbit MQ

Part of https://github.com/Activiti/Activiti/issues/3702

### CODE REVIEW

File: activiti-cloud-messages-service/integration-tests/src/test/java/org/activiti/cloud/messages/integration/tests/rb/MultipleRbMessagesIT.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+@Testcontainers
+class MultipleRbMessagesIT {
+    private static final String INTERMEDIATE_CATCH_MESSAGE_PROCESS = "IntermediateCatchMessageProcess";
+    private static final String INTERMEDIATE_THROW_MESSAGE_PROCESS = "IntermediateThrowMessageProcess";
+    private static final String BUSINESS_KEY = "businessKey";
+
+    private static ConfigurableApplicationContext h2Ctx;
+    private static ConfigurableApplicationContext rbCtx1;
+    private static ConfigurableApplicationContext rbCtx2;
```
Comment (erdemedeiros) [inline activiti-cloud-messages-service/integration-tests/src/test/java/org/activiti/cloud/messages/integration/tests/rb/MultipleRbMessagesIT.java]: Would you mind renaming the variables to `h2Context`, `rbContext1` and `rbContext2`?

File: activiti-cloud-messages-service/integration-tests/src/test/java/org/activiti/cloud/messages/integration/tests/rb/MultipleRbMessagesIT.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+    @Container
+    private static GenericContainer keycloakContainer =
+        new GenericContainer("activiti/activiti-keycloak")
+            .withExposedPorts(8180)
+            .waitingFor(Wait.defaultWaitStrategy());
+
+    @Container
+    private static RabbitMQContainer rabbitMQContainer =
+        new RabbitMQContainer("rabbitmq:3.8.9-management-alpine");
```
Comment (erdemedeiros) [inline activiti-cloud-messages-service/integration-tests/src/test/java/org/activiti/cloud/messages/integration/tests/rb/MultipleRbMessagesIT.java:75]: Is it possible to reuse `RabbitMQContainerApplicationInitializer` and `KeycloakContainerApplicationInitializer`, here?

File: activiti-cloud-runtime-bundle-service/activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/conf/EngineConfigurationIT.java [hunk 3]
```diff
@@ -54,8 +54,55 @@ public void shouldUseCloudActivityBehaviorFactory() {
...
+        assertProperty("spring.cloud.stream.bindings.commandConsumer.destination").isEqualTo("commandConsumer_my-activiti-rb-app");
+        assertProperty("spring.cloud.stream.bindings.commandConsumer.group").isEqualTo("messageConnector");
+        assertProperty("spring.cloud.stream.bindings.commandResults.destination").isEqualTo("commandResults_my-activiti-rb-app");
+
+    }
+
+    @Test
+    public void shouldHaveChannelBindingsSetForSignalEvents() {
+        //when
+        assertProperty("spring.cloud.stream.bindings.signalProducer.destination").isEqualTo("signalEvent");
```
Comment (erdemedeiros) [inline activiti-cloud-runtime-bundle-service/activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/conf/EngineConfigurationIT.java:85]: Should we add the suffix to `signalEvent` as well?

File: activiti-cloud-runtime-bundle-service/activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/conf/EngineConfigurationIT.java [hunk 4]
```diff
@@ -54,8 +54,55 @@ public void shouldUseCloudActivityBehaviorFactory() {
...
+            .as("should have destination set for audit producer")
+            .isEqualTo("engineEvents");
+    }
+
+    @Test
+    public void shouldHaveChannelBindingsSetForMessageEvents() {
+        //when
+        assertProperty("spring.cloud.stream.bindings.messageEvents.destination").isEqualTo("messageEvents_my-activiti-rb-app");
+        assertProperty("spring.cloud.stream.bindings.messageEvents.producer.required-groups").isEqualTo("messageConnector");
+    }
```
Comment (erdemedeiros) [inline activiti-cloud-runtime-bundle-service/activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/conf/EngineConfigurationIT.java:71]: Really nice to have these extra tests here!

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
Some minor details. LGTM otherwise.

File: activiti-cloud-runtime-bundle-service/activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/conf/EngineConfigurationIT.java (hunk 3 above)
Comment (igdianov) [reply activiti-cloud-runtime-bundle-service/activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/conf/EngineConfigurationIT.java:85]: I think it is out of the scope for this PR.  It is better to have a separate issue to advocate changing the signal event behaivor.

File: activiti-cloud-runtime-bundle-service/activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/conf/EngineConfigurationIT.java (hunk 4 above)
Comment (igdianov) [reply activiti-cloud-runtime-bundle-service/activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/conf/EngineConfigurationIT.java:71]: 👍

File: activiti-cloud-messages-service/integration-tests/src/test/java/org/activiti/cloud/messages/integration/tests/rb/MultipleRbMessagesIT.java (hunk 1 above)
Comment (igdianov) [reply activiti-cloud-messages-service/integration-tests/src/test/java/org/activiti/cloud/messages/integration/tests/rb/MultipleRbMessagesIT.java]: Fixed.

File: activiti-cloud-messages-service/integration-tests/src/test/java/org/activiti/cloud/messages/integration/tests/rb/MultipleRbMessagesIT.java (hunk 2 above)
Comment (igdianov) [reply activiti-cloud-messages-service/integration-tests/src/test/java/org/activiti/cloud/messages/integration/tests/rb/MultipleRbMessagesIT.java:75]: I thought about it and tried to use common test containers. It failed. The reason is that application configuration is not SpringBootTest and ContextConfiguration is completely ignored in this case.

**Review by erdemedeiros — APPROVED** [review]
LGMT!

**Pull request merged**